.github/actions/nix_installer_and_cache/action.yml

```diff
@@ -13,38 +13,48 @@ outputs:
     description: Whether the cache was restored or not
     value: ${{ steps.nix-cache.outputs.cache-hit }}
 
-runs:
-  using: composite
-  steps:
-    - name: Cache
-      id: nix-cache
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84
-      with:
-        path: |
-          /tmp/nix-cache
-        key: nix-cache-${{ runner.os }}-${{ github.workflow }}-${{ github.job }}-${{ inputs.cache-unique-id }}-${{ hashFiles('**/flake.lock') }}
-
-    - name: cache hit
-      if: steps.nix-cache.outputs.cache-hit == 'true'
-      shell: bash
-      run: |
-        set -ex
-        sudo mkdir -p /nix
-        sudo chown -R "${USER}:" /nix
-
-        sudo mkdir /etc/nix
-        echo "extra-experimental-features = flakes" > nix.conf
-        echo "extra-experimental-features = nix-command" >> nix.conf
-
-        sudo mv nix.conf /etc/nix/nix.conf
-
-        sudo mv /tmp/nix-cache/** /nix
-
-        echo "/nix/var/nix/profiles/per-user/root/profile/bin" >> $GITHUB_PATH
-        echo "/nix/var/nix/profiles/default/bin" >> $GITHUB_PATH
-
-        sudo chown -R "${USER}:" /nix
-
-    - name: Install Nix
-      if: steps.nix-cache.outputs.cache-hit != 'true'
-      uses: DeterminateSystems/nix-installer-action@3ebd1aebb47f95493b62de6eec0cac3cd74e50a9
+jobs:
+  cache:
+    runs-on: ubuntu-latest
+    steps:
+      - name: restore and save cache
+        id: nix-cache
+        if: ${{ github.event_name == 'schedule' }}
+        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84
+        with:
+          path: |
+            /tmp/nix-cache
+          key: nix-cache-${{ runner.os }}-${{ github.workflow }}-${{ github.job }}-${{ inputs.cache-unique-id }}-${{ hashFiles('**/flake.lock') }}
+
+      - name: cache hit
+        if: steps.nix-cache.outputs.cache-hit == 'true'
+        shell: bash
+        run: |
+          set -ex
+          sudo mkdir -p /nix
+          sudo chown -R "${USER}:" /nix
+  
+          sudo mkdir /etc/nix
+          echo "extra-experimental-features = flakes" > nix.conf
+          echo "extra-experimental-features = nix-command" >> nix.conf
+  
+          sudo mv nix.conf /etc/nix/nix.conf
+  
+          sudo mv /tmp/nix-cache/** /nix
+  
+          echo "/nix/var/nix/profiles/per-user/root/profile/bin" >> $GITHUB_PATH
+          echo "/nix/var/nix/profiles/default/bin" >> $GITHUB_PATH
+  
+          sudo chown -R "${USER}:" /nix
+  
+      - name: Install Nix
+        if: steps.nix-cache.outputs.cache-hit != 'true'
+        uses: DeterminateSystems/nix-installer-action@3ebd1aebb47f95493b62de6eec0cac3cd74e50a9
+
+      - name: restore cache
+        if: ${{ github.event_name != 'schedule' }}
+        uses: actions/cache/restore@704facf57e6136b1bc63b828d79edcd491f0ee84
+        with:
+          path: |
+            /tmp/nix-cache
+          key: nix-cache-${{ runner.os }}-${{ github.workflow }}-${{ github.job }}-${{ inputs.cache-unique-id }}-${{ hashFiles('**/flake.lock') }}
```

Comment:
Let's make this step the second step that runs if `restore and save cache` conditional step doesn't run